src/query/server/query.go

```diff
@@ -119,6 +121,12 @@ type RunOptions struct {
 	// ListenerCh is a programmatic channel to receive the server listener
 	// on once it has opened.
 	ListenerCh chan<- net.Listener
+
+	// CustomHandlers is a list of custom 3rd party handlers.
+	CustomHandlers []options.CustomHandler
+
+	// CustomParseFunction is a custom parsing function.
+	CustomParseFunction promql.ParseFn
```

Comment:
nit: Should we call this `CustomPromQLParseFunction`? In case we need to add another language, etc which may cause a backwards incompat change for those using this code?